pysb/simulator/cuda_ssa.py

```diff
@@ -0,0 +1,297 @@
+from __future__ import print_function
+from pysb.simulator.base import SimulationResult
+from pysb.simulator.ssa_base import SSABase
+import numpy as np
+import os
+import time
+from pysb.pathfinder import get_path
+
+
+try:
+    import pycuda
+    import pycuda.autoinit
+    from pycuda.autoinit import device
+    import pycuda as cuda
+    import pycuda.compiler
+    import pycuda.tools as tools
+    import pycuda.driver as driver
+    import pycuda.gpuarray as gpuarray
+    from mako.template import Template
+except ImportError:
+    pycuda = None
+
+
+class CudaSSASimulator(SSABase):
+    """
+    SSA simulator for NVIDIA gpus
+
+    Use `CUDA_VISIBLE_DEVICES` to set the device.
+
+    Requires `PyCUDA`_, `CUDA`_, and a CUDA compatible NVIDIA gpu.
+
+    .. _PyCUDA :
+        https://documen.tician.de/pycuda/
+    .. _CUDA :
+        https://developer.nvidia.com/cuda-zone
+
+    Parameters
+    ----------
+    model : pysb.Model
+        Model to simulate.
+    tspan : vector-like, optional
+        Time values over which to simulate. The first and last values define
+        the time range. Returned trajectories are sampled at every value unless
+        the simulation is interrupted for some reason, e.g., due to
+        satisfaction
```

Comment:
no tout below, also don't see anything about a logical stopping - is this referring to an earlier version of the code?

Reply:
This is OK - see other instance of this in OpenCLSSASimulator.